src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj

```diff
@@ -119,8 +113,22 @@
     <Compile Include="System\Runtime\CompilerServices\RuleCache.cs" />
     <Compile Include="System\Runtime\CompilerServices\RuntimeOps.cs" />
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition=" '$(IsInterpreting)' != 'true' ">
+    <Compile Include="$(CommonPath)\System\Dynamic\Utils\CollectionExtensions.Map.cs">
+      <Link>Common\System\Dynamic\Utils\CollectionExtensions.Map.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\System\Linq\Expressions\Compiler\AssemblyGen.cs">
+      <Link>Common\System\Linq\Expressions\Compiler\AssemblyGen.cs</Link>
+    </Compile>
+
     <None Include="project.json" />
   </ItemGroup>
+  <ItemGroup Condition=" '$(IsInterpreting)' == 'true' ">
+    <Compile Include="$(CommonPath)\System\Dynamic\Utils\DelegateHelpers.cs">
+      <Link>Common\System\Dynamic\Utils\DelegateHelpers.cs</Link>
+    </Compile>
+
+    <None Include="projectInterpreting.json" />
```

Comment:
@ericstj would have to say for sure but I suspect this isn't doing what you want it to do. I don't believe DNU is restoring based on this json file.